llms/mlx_lm/tokenizer_utils.py

```diff
@@ -254,21 +254,29 @@ class TokenizerWrapper:
     huggingface tokenizer.
     """
 
-    def __init__(self, tokenizer, detokenizer_class=NaiveStreamingDetokenizer):
+    def __init__(
+        self, tokenizer, detokenizer_class=NaiveStreamingDetokenizer, eos_token_ids=None
+    ):
         self._tokenizer = tokenizer
         self._detokenizer = detokenizer_class(tokenizer)
+        self._eos_token_ids = eos_token_ids or [tokenizer.eos_token_id]
```

Comment:
Maybe it is worth making this a set.. for very small # of eos_ids it will probably not matter much,  but safer for long-term.

Reply:
I was just curious so I timed it:

```
eos_ids = [1, 2, 3, 4]
print(timeit.timeit(lambda: -1 in eos_ids, number=100000))
eos_ids = set(eos_ids)
print(timeit.timeit(lambda: -1 in eos_ids, number=100000))
```
Gives:

```
0.0067856249999999965
0.004531958000000003
```

Reply:
Absolutely agree, I have already optimized.